draft-yasskin-dispatch-bundled-exchanges.md

```diff
@@ -0,0 +1,436 @@
+compressed index defined by a future specification) is not present, parsing MUST
+fail.
+
+Return the values returned by parsing each section.
+
+### Parsing the index section {#index-section}
+
+To parse the index, starting at offset *index-start*, with a "responses" section
+starting at *responses-start*, the parser MUST do the following:
+
+Load a CBOR item starting at *index-start*. Parsing MUST fail if any of the
+following is true:
+
+* This item doesn't match the `index` rule in the following CDDL
+  ({{!I-D.ietf-cbor-cddl}}).
+* This item isn't encoded canonically ({{canonical-encoding}}).
+* Any key is not a valid request item ({{request-items}}).
+
+~~~ cddl
+index = {* http-request => [ offset: uint,
+                             length: uint] }
+~~~
+
+Adjust all of the offsets by adding *responses-start*, so they point at the
+start of the response relative to the start of the whole bundle. Note that
+offsets are stored relative to the responses section instead of the start of the
+whole bundle so that they don't need to incorporate their own size, which would
+make it more difficult to encode them canonically.
+
+Interpret the keys as requests as described in {{request-items}}, and return
+this set of requests with each request's adjusted offset and length attached.
+
+### Parsing the manifest section {#manifest-section}
+
+To parse the manifest section, starting at offset *manifest-start*, the parser
+MUST load a CBOR item starting at *manifest-start*. If this item isn't a valid
+request item, as defined by {{request-items}}, the parser MUST fail.
+
+~~~ cddl
+manifest = http-request
+~~~
+
+Return the request represented by this parsed item, interpreted as defined by
+{{request-items}}.
+
+### Parsing the critical section {#critical-section}
+
+To parse the critical section, starting at offset *critical-start*, the parser
+MUST load a CBOR item starting at *critical-start*. If this item doesn't match
+the `critical` rule in the following CDDL ({{!I-D.ietf-cbor-cddl}}), the parser
+MUST fail.
+
+~~~ cddl
+critical = [*tstr]
+~~~
+
+For each value *section-name* in the `critical` list, if the parser does not
+support sections named *section-name*, the parser MUST fail.
+
+This section returns no information.
+
+### Interpreting request CBOR items {#request-items}
+
+HTTP requests are represented by CBOR items matching the following CDDL
+({{!I-D.ietf-cbor-cddl}}).
+
+~~~ cddl
+http-request = {
+  * bstr => bstr
+}
+~~~
+
+A valid request item *R*:
+
+* Matches the above CDDL.
+* Has only lower-case ASCII keys, matching the requirement in Section 8.1.2 of
+  {{?RFC7540}}.
+* Has exactly two keys starting with a ':' character, ':method' and ':url'.
+* *R*\[':method'] is an HTTP method defined as cacheable (Section 4.2.3 of
+  {{!RFC7231}}) and safe (Section 4.2.1 of {{!RFC7231}}), as required for
+  PUSH_PROMISEd requests (Section 8.2 of {{?RFC7540}}).
+* *R*\[':url'] is an absolute URI (Section 4.3 of {{!RFC3986}}).
```

Comment:
It's unclear how to validate this in browsers. There's no such primitive.

Reply:
Can I go with all three? E.g., `http://example.com/%` is not an "absolute URI", but will parse per the URL parser.